compiler/rustc_smir/src/rustc_internal/mod.rs

```diff
@@ -125,18 +136,41 @@ pub fn crate_num(item: &stable_mir::Crate) -> CrateNum {
     item.id.into()
 }
 
+// A thread local variable that stores a pointer to the tables mapping between TyCtxt
+// datastructures and stable MIR datastructures
+scoped_thread_local! (static TLV: Cell<*const ()>);
+
+pub(crate) fn init<'tcx>(tables: &TablesWrapper<'tcx>, f: impl FnOnce()) {
+    assert!(!TLV.is_set());
+    let ptr: *const () = &tables as *const &_ as _;
```

Comment:
This takes a reference to `tables` which is already a reference

Reply:
Indeed. I was just replicating the same logic as our current TLV implementation:
https://github.com/rust-lang/rust/blob/41aa06ecf963cf3529146159f761d6e5b5f9cb48/compiler/stable_mir/src/lib.rs#L249